### PR TITLE
fix execution even if guard_file already exists

### PR DESCRIPTION
**Pull request by Stromweld**
this should fix idempotency with the guard_file

### CODE REVIEW

File: recipes/regiis.rb [hunk 1]
```diff
@@ -28,4 +28,5 @@
   command "%WINDIR%\\Microsoft.Net\\Framework64\\v4.0.30319\\aspnet_regiis.exe -i -enable > #{guard_file}"
   creates guard_file
```
Comment (Annih) [inline recipes/regiis.rb]: I invite you to read the documentation about the `creates` property:
https://docs.chef.io/resource_execute.html#properties

File: recipes/regiis.rb [hunk 2]
```diff
@@ -28,4 +28,5 @@
...
   only_if 'sc.exe query W3SVC'
+  not_if { File.exist?(guard_file) }
```
Comment (Annih) [inline recipes/regiis.rb]: You are replicating the `creates` behavior

File: recipes/regiis.rb (hunk 1 above)
Comment (Stromweld) [inline recipes/regiis.rb]: I may be wrong, but seeing the name guard_file my thought was that it's intended to keep the command from running on every chef-client run. I assume it's meant to run once create the file and then not run again as long as the file exists. If my assumption is correct then this is not working as intended as it runs this command everytime chef-client runs even with the file in place.

File: recipes/regiis.rb (hunk 2 above)
Comment (Stromweld) [inline recipes/regiis.rb]: This doesn't actually replicate the same behavior. The not_if stops the entire execute resource from running where the creates doesn't stop the command or resource from running but only stops it from overwriting the file in place if it already exists. The command still runs with every chef-client run. 

The link you gave above doesn't say anything about preventing command from running just suppressing it's output to the filesystem when it does. Here is a quote from the link:

creates
Ruby Type: String

Prevent a command from creating a file when that file already exists.

File: recipes/regiis.rb (hunk 1 above)
Comment (Annih) [inline recipes/regiis.rb]: Ok, maybe the documentation is not clear enough.
Take a look at the implementation: https://github.com/chef/chef/blob/master/lib/chef/provider/execute.rb#L51

If the file is present the execute resource does nothing.

What make you say that the execute resource run at every chef-client runs? Could you provide a chef-client output output?

File: recipes/regiis.rb (hunk 2 above)
Comment (Stromweld) [inline recipes/regiis.rb]: I'm sorry, you are right, I was mistaking the output from the only_if that shows up in the chef-client run as the command running but just realized it's the the query from the only_if part.

screen shot 2017-05-04 at 11 12 11 am

**Comment by Annih** [comment]
Ok great :)

**Pull request closed without merging**